addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -53,21 +53,36 @@ class ChannelMember(models.Model):
     @api.autovacuum
     def _gc_unpin_outdated_sub_channels(self):
         outdated_dt = fields.Datetime.now() - timedelta(days=2)
-        domain = expression.AND(
-            [
-                [
-                    ("channel_id.parent_channel_id", "!=", False),
-                    ("last_interest_dt", "<", outdated_dt),
-                ],
-                expression.OR(
-                    [
-                        [("channel_id.last_interest_dt", "=", False)],
-                        [("channel_id.last_interest_dt", "<", outdated_dt)],
-                    ]
-                ),
-            ]
+        self.env["discuss.channel"].flush_model()
+        self.env["discuss.channel.member"].flush_model()
+        self.env["mail.message"].flush_model()
+        self.env.cr.execute(
+            """
+            SELECT member.id
+              FROM discuss_channel_member member
+              JOIN discuss_channel channel
+                ON channel.id = member.channel_id
+             WHERE (
+                       member.unpin_dt IS NULL 
+                    OR member.last_interest_dt >= member.unpin_dt
+                    OR channel.last_interest_dt >= member.unpin_dt
+               )
+               AND COALESCE(member.last_interest_dt, member.create_date) < %(outdated_dt)s
+               AND COALESCE(channel.last_interest_dt, channel.create_date) < %(outdated_dt)s
```

Comment:
Why do we need to `COALESCE` on `last_interest_dt` when checking `outdated_dt` but not above when checking `unpin_dt`?